src/libraries/System.Linq/src/System/Linq/OrderedEnumerable.cs

```diff
@@ -16,6 +16,8 @@ internal abstract partial class OrderedEnumerable<TElement> : IOrderedEnumerable
 
         private int[] SortedMap(Buffer<TElement> buffer) => GetEnumerableSorter().Sort(buffer._items, buffer._count);
 
+        private void SortElements(Buffer<TElement> buffer) => GetEnumerableSorter().SortElements(buffer._items, buffer._count);
```

Comment:
I would use a name that emphasizes that this is an in-place sort, i.e. `buffer` is going to get mutated.